src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -369,29 +392,53 @@ def _purgeTree(self, ctx, baseDirPfn):
         :param ctx:  The gfal2 context object
         :return:     Bool: True if it managed to purge everything, False otherwise
         """
+        # NOTE: It deletes only directories and does not try to unlink any file.
+
         successList = []
 
+        # First test if baseDirPfn is actually a directory entry:
+        try:
+            entryStat = ctx.stat(baseDirPfn)
+        except gfal2.GError as gfalExc:
+            e = sys.exc_info()[1]
```

Comment:
This line we discussed over zoom yesterday. If you managed to find out that manipulating the `entryStat` object is enough, without this sys.exc_info call, then I would be in favor of that.